lua/spectre_common.lua

```diff
@@ -325,6 +361,37 @@ local function normalize_uri(uri)
     return uri_path .. '?' .. sorted_params
 end
 
+-- Normalizes the request body by addding only vary fields
+-- and sorting the keys in lexicographical order
+local function normalize_body(request_body, cache_entry)
+    -- Normalize body only for POST methods.
+    if cache_entry.request_method ~= 'POST' then
+        return nil
+    end
+
+    -- If there is no body, nothing to normalize.
+    if request_body == nil then
+        return nil
+    end
+
+    local body = json:decode(request_body)
+    local keys = {}
```

Comment:
why don't you directly initialize the map with `table.insert(keys, cache_entry.post_body_id)`?

Reply:
Fixed. Initializing it as 
`local keys = {cache_entry.post_body_id}`